python/pyspark/core/rdd.py

```diff
@@ -5370,6 +5370,18 @@ def _test() -> None:
     import tempfile
     from pyspark.core.context import SparkContext
 
+    try:
+        # Numpy 2.0+ changed its string format,
+        # adding type information to numeric scalars.
+        import numpy as np
```

Comment:
Here actually did a try catch but I think there's some issue related to import ..

Reply:
Let's move this to sql and ml only for now because both modules use `numpy`.

Reply:
Thank you. 

Yes, the problem is that `try-catch` didn't handle `ModuleNotFoundError` and causes failure like the following.
```
$ python/run-tests.py --python-executables python3 --modules pyspark-core
...
  File "/Users/dongjoon/APACHE/spark-merge/python/pyspark/core/rdd.py", line 5376, in _test
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
```

Reply:
> Let's move this to sql and ml only for now because both modules use `numpy`.

+1 for moving.